src/lib/xof/shake_xof/shake_xof.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * SHAKE-128 and SHAKE-256
+ * (C) 2016-2023 Jack Lloyd
+ *     2022-2023 Fabian Albert, Michael Boric, René Meusel - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#include <botan/internal/shake_xof.h>
+
+#include <botan/exceptn.h>
+#include <botan/internal/loadstor.h>
+
+namespace Botan {
+
+SHAKE_XOF::SHAKE_XOF(size_t capacity) : m_keccak(capacity, 0xF, 4), m_output_generated(false) {
+   BOTAN_ASSERT_NOMSG(m_keccak.byte_rate() >= 72 && m_keccak.byte_rate() <= 168);
+}
+
+void SHAKE_XOF::clear() {
+   m_keccak.clear();
+   m_output_generated = false;
+}
+
+void SHAKE_XOF::add_data(const uint8_t key[], size_t length) {
```

Comment:
Calling this param `key` seems odd since there is no requirement that XOF input be secret, and is inconsistent with the other declarations which call this `input` which seems a better name to capture the non-specific nature of the bytes here.

Reply:
For sure. Not sure where this came from. Perhaps a copy & paste quirk. Good catch.